odoo_tools/utils/proj.py

```diff
@@ -25,3 +29,65 @@ def get_current_version(serie_only=False):
     if serie_only:
         ver = ver.split(".")[0]
     return ver
+
+
+def setup_venv(venv_dir, odoo_src_path=None):
+    venv_dir = build_path(venv_dir)
+    if os.path.isfile(os.path.join(venv_dir, "pyenv.cfg")):
+        ui.echo(f"Reusing existing venv {venv_dir}")
+    else:
+        venv.create(venv_dir, with_pip=True)
+    pip = os.path.join(venv_dir, "bin", "pip")
+    if odoo_src_path is None:
+        odoo_src_path = build_path(get_conf_key("odoo_src_rel_path"))
+
+    if not os.path.isfile(os.path.join(venv_dir, "bin", "odoo")):
+        subprocess.call(
+            [pip, "install", "-r", os.path.join(odoo_src_path, "requirements.txt")]
+        )
+        subprocess.call([pip, "install", "setuptools-odoo"])
```

Comment:
IMO we should generate a local-requirement.txt file and add all these deps there. Could you add a TODO?